scipy/ndimage/src/_rank_filter_1d.cpp

```diff
@@ -180,7 +181,13 @@ void _rank_filter(T *in_arr, int rank, int arr_len, int win_len, T *out_arr,
     }
     break;
   case WRAP:
-    for (i = arr_len - lim - 1 - 2 * origin; i < arr_len; i++) {
+    if (win_len % 2 == 0) {
+        offset = 2;
+    }
+    else {
+        offset = 0;
+    }
```

Comment:
this block is based on an analysis of the indices visited by `median_filter` in the default `reflect` mode because of the patterning similarities between the two approaches from the docs:

> ‘reflect’ d c b a | a b c d | d c b a
> ‘wrap’  a b c d | a b c d | a b c d

kind of a "flipped" order to boundary handling.

This is a bit of a fishing expedition though.

Reply:
One point of caution--with the memory initialization issue repaired, I no longer need this patch, though the patch is also not harmful with the current suite of tests that we have.

Objectively that probably means that we should remove this part of the shim. That said, it may also be sensible to expand the testing to include non-zero `origin` values vs. `1.14.1` and n-dims vs. `1.14.1` because the lack of sensitivity on the indexing here makes me a bit suspicious.